Ghpr.Core/Reporter.cs

```diff
@@ -1,10 +1,10 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Threading;
 using Ghpr.Core.Common;
 using Ghpr.Core.EmbeddedResources;
 using Ghpr.Core.Extensions;
 using Ghpr.Core.Interfaces;
-using Ghpr.Core.Utils;
 
 namespace Ghpr.Core
```

Comment:
ActionHelper seems redundant. Handling concurrency issues is the responsibility of concrete objects dealing with shared resources.